rbac_api.go

```diff
@@ -51,36 +50,66 @@ func (e *Enforcer) HasRoleForUser(name string, role string) (bool, error) {
 
 // AddRoleForUser adds a role for a user.
 // Returns false if the user already has the role (aka not affected).
-func (e *Enforcer) AddRoleForUser(user string, role string) (bool, error) {
-	return e.AddGroupingPolicy(user, role)
+func (e *Enforcer) AddRoleForUser(user string, role string, domain ...string) (bool, error) {
+	if len(domain) == 0 {
+		return e.AddGroupingPolicy(user, role)
```

Comment:
Do we have to repeat the code for each function? Is there any way that we could first populate the parameters? Like:

```go
if len(domain) == 0 {
	args = [user, role]
else if len(domain) == 1 {
	args = [user, role, domain[0]]

return e.AddGroupingPolicy(args)
```